lib/service/base.rb

```diff
@@ -261,24 +274,22 @@ def initialize(exceptions, &block)
         instance_exec(&block)
       end
 
-      def call(instance, context)
-        context[result_key] = Context.build
+      def run_step
         steps.each do |step|
           @current_step = step
           step.call(instance, context)
         end
       rescue *exceptions => e
         raise e if e.is_a?(Failure)
         context[@current_step.result_key].fail(raised_exception?: true, exception: e)
-        context[result_key].fail(exception: e)
+        context[result_key][:exception] = e
```

Comment:
`#fail` is not called on the `TryStep` result object anymore, and its outcome is handled by checking whether an `exception` is present. The main reason for this small internal change is to be able to display the whole `try` block in the steps inspector output.